specification/botservice/resource-manager/Microsoft.BotService/preview/2018-07-12/botservice.json

```diff
@@ -1012,6 +1012,283 @@
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            },
+            "description": "The parameters to provide for the new Enterprise Channel."
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "If resource is created successfully or already existed, the service should return 200 (OK).",
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            }
+          },
+          "201": {
+            "description": "If resource is created successfully, the service should return 201 (Created). Execution to continue asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Enterprise Channel"
+        ],
+        "description": "Updates an Enterprise Channel.",
+        "operationId": "EnterpriseChannels_Update",
+        "x-ms-examples": {
+          "Create Bot": {
+            "$ref": "./examples/UpdateEnterpriseChannel.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "x-ms-client-flatten": true,
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            },
+            "description": "The parameters to provide to update the Enterprise Channel."
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "If resource is created successfully or already existed, the service should return 200 (OK).",
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            }
+          },
+          "201": {
+            "description": "If resource is created successfully, the service should return 201 (Created). Execution to continue asynchronously.",
```

Comment:
Does this mean that you're conforming to LRO semantics? https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-long-running-operation

Reply:
We will do that. I think I need to add a route to handle the LRO responses.

Reply:
@shahabhijeet, do you have any insight to offer to @NickEricson on Long Running Operations implementation?

Reply:
@marstr @shahabhijeet Is there anything else we need to do here?